### PR TITLE
fix itests

### PR DESCRIPTION
**Pull request by pwinckles**
This PR hopefully fixes a couple of ITests that intermittently fail. No promises.

A successful run: https://github.com/pwinckles/fcrepo4/actions/runs/451290244

### CODE REVIEW

File: fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/FedoraVersioningIT.java [hunk 1]
```diff
@@ -1021,9 +1021,15 @@ public void testDatetimeNegotiationLDPRv() throws Exception {
 
         final CloseableHttpClient customClient = createClient(true);
 
-        final String uri = createVersionedContainer(id);
+        createVersionedContainer(id);
+
+        final String version1Uri;
 
-        final String version1Uri = createMemento(uri);
+        try (final CloseableDataset dataset = getDataset(getObjMethod(id + "/" + FCR_VERSIONS))) {
```
Comment (awoods) [inline fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/FedoraVersioningIT.java:1028]: Given the number of times the following is implemented in this test class:
```
final String versionUri = createMemento(uri);
```
..it is conceivable that pulling this new logic into a new `private` method could be useful.

File: fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/TestIsolationExecutionListener.java [hunk 2]
```diff
@@ -32,7 +38,24 @@
...
-        ocflRepo.listObjectIds().forEach(ocflRepo::purgeObject);
+        final var ocflConfig = getBean(testContext, OcflPropsConfig.class);
+
+        final var hasError = new AtomicBoolean(false);
+
+        ocflRepo.listObjectIds().forEach(object -> {
+            try {
+                ocflRepo.purgeObject(object);
+            } catch (RuntimeException e) {
+                // Recursive deletes don't behave well on Windows and it's possible for the above to error out.
```
Comment (awoods) [inline fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/TestIsolationExecutionListener.java:49]: Did you forget to toggle `hasError` when catching an exception?

**Review by awoods — CHANGES_REQUESTED** [review]

File: fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/TestIsolationExecutionListener.java (hunk 2 above)
Comment (pwinckles) [reply fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/TestIsolationExecutionListener.java:49]: whoops

**Comment by pwinckles** [comment]
Looks like the new Windows logic needs some more work: https://github.com/pwinckles/fcrepo4/runs/1626984734?check_suite_focus=true

Will fix next week.

**Review by awoods — DISMISSED** [review]

**Comment by pwinckles** [comment]
@awoods Don't merge this. The Windows code is still broken. I'll fix it on Monday

**Comment by pwinckles** [comment]
I think this is good to go now. The most recent run triggered the Windows failure case and it successfully recovered.

https://github.com/fcrepo/fcrepo/runs/1644170923?check_suite_focus=true#step:6:5887

**Review by whikloj — APPROVED** [review]
👍

**Pull request merged**